catalog/mgmp/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/mgmp/MgmpConverter.java

```diff
@@ -0,0 +1,1249 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.UUID;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.xml.ws.Holder;
+import org.apache.commons.collections.BidiMap;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.codice.alliance.catalog.core.api.types.Isr;
+import org.codice.alliance.catalog.core.api.types.Security;
+import org.codice.ddf.spatial.ogc.csw.catalog.common.GmdConstants;
+import org.codice.ddf.spatial.ogc.csw.catalog.converter.AbstractGmdConverter;
+import org.codice.ddf.spatial.ogc.csw.catalog.converter.XstreamPathValueTracker;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Convert a metacard to MGMPv2 compliant XML.
+ *
+ * <p>Not thread safe.
+ */
+public class MgmpConverter extends AbstractGmdConverter {
+
+  private static final String BOUNDING_BOX_FORMAT = "%.15f";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MgmpConverter.class);
+
+  private static final int MIN_CLOUD_COVERAGE = 0;
+
+  private static final int MAX_CLOUD_COVERAGE = 100;
+
+  private static final String ID_REGEX = "^.{32}$";
+
+  private static final int ID_SEGMENT_1_START = 0;
+
+  private static final int ID_SEGMENT_1_END = 8;
+
+  private static final int ID_SEGMENT_2_START = 8;
+
+  private static final int ID_SEGMENT_2_END = 12;
+
+  private static final int ID_SEGMENT_3_START = 12;
+
+  private static final int ID_SEGMENT_3_END = 16;
+
+  private static final int ID_SEGMENT_4_START = 16;
+
+  private static final int ID_SEGMENT_4_END = 20;
+
+  private static final int ID_SEGMENT_5_START = 20;
+
+  private static final int ID_SEGMENT_5_END = 32;
+
+  private static final String ID_SEPARATOR = "-";
+
+  private static final Pattern CRS_FILTER_PATTERN = Pattern.compile("^[^:]+:[^:]+$");
+
+  private Predicate<String> crsFilter = s -> CRS_FILTER_PATTERN.matcher(s).matches();
+
+  private int geographicElementIndex = 1;
+
+  private int dateElementIndex = 1;
+
+  private int dataIdentificationIndex = 1;
+
+  private int resourceContraintsIndex = 1;
+
+  private Supplier<String> gmlIdSupplier = this::generateGmlGuid;
+
+  private XstreamPathValueTracker pathValueTracker;
+
+  private MetacardImpl metacard;
+
+  private BidiMap securityMappingList;
+
+  private String sourceSystemName;
+
+  private BidiMap inverseMappingList;
+
+  private static final String SECURITY_MAPPING_LIST_KEY = "securityMappingList";
+
+  private static final String SOURCE_SYSTEN_NAME_KEY = "sourceSystemName";
+
+  public void refresh(Map<String, Object> properties) {
```

Comment:
❓ What happens if this refreshes while transforms are being run? Should it be synchronized with the entry point?